.github/workflows/test_weaver-docker-build.yaml

```diff
@@ -37,7 +25,16 @@ jobs:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v3.5.2
 
-      - name: Build Image
+      - uses: dorny/paths-filter@v2.11.1
+        id: changes
+        with:
+          filters: |
+            weaver_code_changed:
+              - './weaver/**!(*.md|*.css|*.html|*.jpg|*.jpeg|*.png)'
```

Comment:
These are unit tests, we can be more specific to paths here.
Since `paths` were one time specified for each workflow I had to give whole weaver directory, but here we are specifying for each job, we can be more specific, do you want me to edit these so that they can be more efficient?